### PR TITLE
project duration fixed 

### PR DESCRIPTION
**Pull request by mahigangal**
## 📝 Description

Fixed project duration estimation to handle missing timestamps, avoid zero‑length “now” fallbacks, and provide a human‑readable duration string without microseconds. Cleaned up imports and small docstring/comment clarity in analysis service.

Users were seeing durations like 0:00:00.006828 when analyzing a project. The updated estimator now skips missing dates, uses the earliest modified date when created is missing (and vice versa), and outputs a readable string (e.g., “1 day, 3 hours”).

**Closes:** #350 

---

## 🔧 Type of Change

- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

python -m unittest discover -s test -p "test_project_duration.py"

Added/updated unit tests covering fallback logic and human‑readable formatting.

---

## ✓ Checklist

- [ ] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> 

<details>
<summary>Click to expand screenshots</summary>

Screenshot 2026-01-31 at 9 35 58 PM

</details>

### CODE REVIEW

**Review by Puneet-Maan — CHANGES_REQUESTED** [review]
thanks for taking the time to look into this, Mahi!
I think the duration calculation breaks when some files only have created time and others only have modified time. It picks start from created only and end from modified only, so it can give a wrong or even negative duration.

I think what we can do is maybe use the earliest timestamp available per file for start, and the latest timestamp available per file for end, but please have a look.

File: src/core/analysis_service.py [hunk 1]
```diff
@@ -134,8 +131,8 @@ def analyze_project(root: Path, use_ai_analysis=False) -> None:
     """
 
     display_name = root.name
-    hierarchy = FileMetadataExtractor(root).file_hierarchy()  #Metadata extracted and datetime objects converted to strings
-    duration = str(Project_Duration_Estimator(hierarchy).get_duration()) #Project duration estimate
+    hierarchy = FileMetadataExtractor(root).file_hierarchy()  #Metadata extracted with datetime objects
+    duration = Project_Duration_Estimator(hierarchy).get_duration_human() #Project duration estimate
```
Comment (ThunderIW) [inline src/core/analysis_service.py]: Good use of the new function you created here

File: src/core/project_duration_estimation.py [hunk 2]
```diff
@@ -56,28 +80,29 @@ def __list_dates_recurse(self, node: dict):
...
         '''
-        Function that takes a file from the recursive method list_dates_recursive and extracts the creation and last modified dates of files
+        Extracts earliest and latest timestamps for a single file.
         '''
-        self.created_dates.append(file["created"])
-        self.mod_dates.append(file["modified"])
+        created = file.get("created")
+        modified = file.get("modified")
+        
+        timestamps = [t for t in (created, modified) if t is not None]
```
Comment (ThunderIW) [inline src/core/project_duration_estimation.py]: Nice — this cleanly handles the mixed-timestamp case (file has only created or only modified).

**Review by ThunderIW — APPROVED** [review]
Looks good – no code-breaking issues. The tuple solution for file_ranges handles mixed/missing timestamps well, and the human-readable formatter is robust against edge cases. Good test coverage, too.

Made a few minor suggestions (inline comments, `@staticmethod`, catching ValueError instead of bare Exception, and a test case for mixed timestamps) but nothing critical. PR is approved .

Note: I am acting as the second Reviewer since @Puneet-Maan Reviewed it first 
 and @mahigangal  has made changes on what Puneet has said

**Review by Puneet-Maan — APPROVED** [review]
thank yiou for fixing the above issues. approving
